server/game/gamesteps/uiprompt.js

```diff
@@ -14,14 +14,25 @@ class UiPrompt extends BaseStep {
 
     complete() {
         this.completed = true;
+        if(this.getPlayer()) {
+            if(this.getPlayer().stopClock) {
```

Comment:
Are the checks like this necessary? I would think that a player object would always have a `stopClock` and `startClock` methods.

Reply:
I found out that some of the player objects created during tests are not "real" player objects and only have specific methods. Those tests fail without those checks in place. Another way of fixing it would be to change the creation of the player objects in those particular tests. What would you recommend?

Reply:
The test objects are mocks, if you want them to have those methods you need to add it to the mock. Shouldn't be too difficult to set up

Reply:
If i understand correctly, this code creates the mock for a player object: jasmine.createSpyObj('player' ...)
Searching for instances of this code, i find 32 results. Is there an easier way to add the methods to the mocks than adding stop and startClock to each of those 32 mock creation calls? Maybe a helper method for creating player mocks with default methods in addition to those that are added as a parameter?

Reply:
It should only be added where it's needed, if that happens to be all 32 then that's unfortunate :) But yes in that case it would probably make sense to refactor the thing to all get created in one place

Reply:
Fixed it now for the instances where it was needed.